src/plonk_recursion.rs

```diff
@@ -558,9 +697,21 @@ fn halo_g_recursive<C: HaloCurve>(
     let mut x_power = x;
     for &u_i in us.iter().rev() {
         let u_i_inv = builder.inv(u_i);
-        let term = builder.mul_add(u_i_inv, x_power, u_i);
+        let term = builder.mul_add(u_i, x_power, u_i_inv);
```

Comment:
Was this causing an issue? I think the old version matches the convention in the Halo paper.

Reply:
So I think there's a mistake in the Halo paper and they've swapped `u_i` and `u_i^-1` in the definition of the `g` polynomial. For example, the 0-th element in the `s` vector is `prod u_i^-1`, while the 0-th degree term of `g` is `prod u_i`.

Reply:
Good point!